File: android/lib/net/response/response.dart
import 'package:android/net/common/message.dart';
import 'package:json_annotation/json_annotation.dart';

part 'response.g.dart';

@JsonSerializable()
class Response extends Message {
  @JsonKey(name: "Status")
  final Status c_status;
  @JsonKey(name: "Message")
  final String c_message;
  @JsonKey(name: "About")
  final About c_about;

  Response(
    super.c_type,
    super.c_userAgent,
    this.c_status,
    this.c_message, {
    this.c_about = About.defaultValue,
  });

  Status get status => c_status;
  String get message => c_message;
  About get about => c_about;

  set c_type(Type p_type) => c_type = p_type;
}

enum Status {
  @JsonValue("SUCCESS")
  success,
  @JsonValue("FAILURE")
  failure,
}

enum About {
  @JsonValue("DEFAULT")
  defaultValue,
  @JsonValue("FILTER_STORES_REQUEST")
  filterStores,
}

File: app/src/main/java/com/github/giorgosathanasopoulos/distributed_systems_aueb/android/backend/network/response/Response.java
package com.github.giorgosathanasopoulos.distributed_systems_aueb.android.backend.network.response;

import com.github.giorgosathanasopoulos.distributed_systems_aueb.android.backend.network.Message;
import com.google.gson.annotations.SerializedName;

import com.github.giorgosathanasopoulos.distributed_systems_aueb.android.backend.json.JsonUtils;

public class Response extends Message {

    @SerializedName("Status")
    private final Status c_Status;

    @SerializedName("Message")
    private final String c_Message;

    @SerializedName("About")
    private final About c_About;

    public Response(
            UserAgent p_UserAgent,
            int p_Id,
            Status p_Status,
            String p_Message) {
        super(Type.RESPONSE, p_Id, p_UserAgent);
        this.c_Status = p_Status;
        this.c_Message = p_Message;
        this.c_About = About.DEFAULT;
        setSrc(JsonUtils.toJson(this));
    }

    public Response(
            UserAgent p_UserAgent,
            int p_Id,
            Status p_Status,
            String p_Message,
            About p_About) {
        super(Type.RESPONSE, p_Id, p_UserAgent);
        this.c_Status = p_Status;
        this.c_Message = p_Message;
        this.c_About = p_About;
        setSrc(JsonUtils.toJson(this));
    }

    public enum Status {
        SUCCESS,
        FAILURE,
    }

    public enum About {
        DEFAULT,
        LIST_STORES_REQUEST,
        FILTER_STORES_REQUEST,
        LIST_PRODUCTS_REQUEST,
        SHOW_SALES_FOOD_TYPE_REQUEST,
        SHOW_SALES_STORE_TYPE_REQUEST,
    }

    public Status getStatus() {
        return c_Status;
    }

    public String getMessage() {
        return c_Message;
    }

    public About getAbout() {
        return c_About;
    }
}
